src/utils/transactorHelper.ts

```diff
@@ -0,0 +1,51 @@
+import { Contract } from '@ethersproject/contracts'
+import { Transactor, TransactorOptions } from 'hooks/Transactor'
+import { isUndefined } from 'lodash'
+import invariant from 'tiny-invariant'
+
+export function handleTransactor(props: {
+  fnName: string
+  args: unknown[]
+  version: 'v1' | 'v2'
+  transactor?: Transactor
+  contract?: Contract
+  txOpts?: Omit<TransactorOptions, 'value'>
+}): Promise<boolean> {
+  const { transactor, contract, fnName, txOpts, version, args } = props
+
+  const params = Object.keys(props) as ReturnType<() => keyof typeof props>[]
+  const missingParam = params.find(param => !props[param])
+  const argParams = args.filter(arg => !isUndefined(arg)).join(', ')
+
```

Comment:
I manually tested this a couple of times. But, I can probably add some unit tests now that we have it in a function.

Reply:
Logic here isn't right, moving to draft